Please port to numpy-2

##### Is your feature request related to a problem?
<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]
 -->
Please port to numpy-2. Some distros (e.g. Fedora) no more provide numpy<2.

##### Describe the solution you'd like
<!-- A clear and concise description of what you want to happen. -->
The GUI works with numpy-2.

##### Describe alternatives you've considered
<!-- A clear and concise description of any alternative solutions or features you've considered. -->
No way to distro package it to fedora-42, because there is no numpy-1

##### Additional context
<!-- Add any other context or screenshots about the feature request here. -->
numpy-2 migration guide: https://numpy.org/devdocs/numpy_2_0_migration_guide.html
Maybe I could port it myself, but it will take time, so at least reporting it for the RFE to be tracked.

With the following patch:
```
--- a/src/urh/cythonext/signal_functions.pyx
+++ b/src/urh/cythonext/signal_functions.pyx
@@ -238,9 +238,10 @@ cdef np.ndarray[np.float32_t, ndim=1] gauss_fir(float sample_rate, uint32_t samp
                                                 int(filter_width * samples_per_symbol) + 1,
                                                 dtype=np.float32)
     cdef float ts = samples_per_symbol / sample_rate  # symbol time
-    cdef np.ndarray[np.float32_t] h = np.sqrt((2 * np.pi) / (np.log(2))) * bt / ts * np.exp(
+    cdef np.ndarray[np.double_t] h = np.sqrt((2 * np.pi) / (np.log(2))) * bt / ts * np.exp(
         -(((np.sqrt(2) * np.pi) / np.sqrt(np.log(2)) * bt * k / samples_per_symbol) ** 2))
-    return h / h.sum()
+    cdef np.ndarray[np.float32_t] hh = h.astype(np.float32)
+    return hh / hh.sum()
 
 cdef float clamp(float x) nogil:
     if x < -1.0:
```
tests results: 1 failed, 341 passed

The patch is far from being optimal any hints for optimization are welcome.

Regarding the remaining 1 failure, it's in:
```
self = <tests.awre.test_generated_protocols.TestGeneratedProtocols testMethod=test_without_preamble>

    def test_without_preamble(self):
        alice = Participant("Alice", address_hex="24")
        broadcast = Participant("Broadcast", address_hex="ff")
    
        mb = MessageTypeBuilder("data")
        mb.add_label(FieldType.Function.SYNC, 16)
        mb.add_label(FieldType.Function.LENGTH, 8)
        mb.add_label(FieldType.Function.SRC_ADDRESS, 8)
        mb.add_label(FieldType.Function.SEQUENCE_NUMBER, 8)
    
        pg = ProtocolGenerator(
            [mb.message_type],
            syncs_by_mt={mb.message_type: "0x8e88"},
            preambles_by_mt={mb.message_type: "10" * 8},
            participants=[alice, broadcast],
        )
    
        for i in range(20):
            data_bits = 16 if i % 2 == 0 else 32
            source = pg.participants[i % 2]
            destination = pg.participants[(i + 1) % 2]
            pg.generate_message(
                data="1010" * (data_bits // 4), source=source, destination=destination
            )
    
        # self.save_protocol("without_preamble", pg)
        self.clear_message_types(pg.messages)
        ff = FormatFinder(pg.messages)
        ff.known_participant_addresses.clear()
    
        ff.run()
        self.assertEqual(len(ff.message_types), 1)
    
        mt = ff.message_types[0]
        sync = mt.get_first_label_with_type(FieldType.Function.SYNC)
        self.assertEqual(sync.start, 0)
        self.assertEqual(sync.length, 16)
    
        length = mt.get_first_label_with_type(FieldType.Function.LENGTH)
>       self.assertEqual(length.start, 16)
E       AttributeError: 'NoneType' object has no attribute 'start'

tests/awre/test_generated_protocols.py:85: AttributeError
```
and here I have no idea what's wrong, any hints?

If we come up with complete solution, I am OK to create PR.

--- a/data/requirements.txt
+++ b/data/requirements.txt
@@ -1,4 +1,4 @@
-numpy<2.0.0
+numpy>=2.0,<3.0
 pyqt5
 psutil
 cython<3.1

--- a/setup.py
+++ b/setup.py
@@ -142,7 +142,7 @@ def read_long_description():
         return ""
 
 
-install_requires = ["numpy<2.0.0", "psutil", "cython<3.1", "setuptools"]
+install_requires = ["numpy>=2.0,<3.0", "psutil", "cython<3.1", "setuptools"]
 if IS_RELEASE:
     install_requires.append("pyqt5")
 else:
@@ -165,7 +165,7 @@ setup(
     license="GNU General Public License (GPL)",
     download_url="https://github.com/jopohl/urh/tarball/v" + str(version.VERSION),
     install_requires=install_requires,
-    setup_requires=["numpy<2.0.0"],
+    setup_requires=["numpy>=2.0,<3.0"],
     python_requires=">=3.9",
     packages=get_packages(),
     ext_modules=get_extensions(),

--- a/src/urh/awre/FormatFinder.py
+++ b/src/urh/awre/FormatFinder.py
@@ -492,7 +492,7 @@ class FormatFinder(object):
                 possible_solutions,
                 key=lambda sol: (
                     sum(r.score for r in sol),
-                    -sum(r.length_in_bits for r in sol),
+                    -sum(int(r.length_in_bits) for r in sol),
                     "length" in {r.field_type for r in sol},
                     "".join(r.field_type[0] for r in sol),
                 ),

